src/consumer.c

```diff
@@ -34,6 +41,7 @@ consumer_free(Consumer *c)
 {
     if ((*c) == NULL)
         return;
+    hook_free((*c)->preadd);
```

Comment:
Please add an empty line above, to make it more clear that this line is free-standing.